concrete-core/src/specification/engines/lwe_bootstrap_key_creation.rs

```diff
@@ -14,6 +14,25 @@ engine_error! {
                               the precision of the ciphertext."
 }
 
+impl<EngineError: std::error::Error> LweBootstrapKeyCreationError<EngineError> {
+    pub fn perform_generic_checks(
+        decomposition_base_log: DecompositionBaseLog,
+        decomposition_level_count: DecompositionLevelCount,
+        integer_precision: usize,
+    ) -> Result<(), Self> {
+        if decomposition_base_log.0 == 0 {
+            return Err(Self::NullDecompositionBaseLog);
+        }
+        if decomposition_level_count.0 == 1 {
+            return Err(Self::NullDecompositionLevelCount);
```

Comment:
Here I hadn't noticed that for l=1 the error is NullDecopositionLevelCount. This looks strange, no?

Reply:
Yeah, naming is not that great, maybe we should use `UnitDecompositionLevelCount` ? Or `SingleDecompositionLevelCount` ? Or `OneDecompositionLevelCount` ? This will be a breaking change though.

Reply:
I think we can afford this breaking change at this stage, no?

Reply:
Looking further, the test should verify that the value is different from zero, so we don't have to change the name, but the test should be fixed.